packages/node/src/login/oidc/incomingRedirectHandler/AuthCodeRedirectHandler.ts

```diff
@@ -169,7 +170,7 @@ export class AuthCodeRedirectHandler implements IIncomingRedirectHandler {
     }
     const authFetch = await buildAuthenticatedFetch(tokenSet.access_token, {
       dpopKey,
-      refreshOptions,
+      refreshOptions: oidcContext.keepAlive ? refreshOptions : undefined,
```

Comment:
This is a key part of this PR: is `keepAlive` is set to false, we initialize the authenticated fetch as if there was no refresh token, so that it doesn't set a timeout to refresh the access token before it expires.

This part of the code runs at the point of the token request in the authorization code flow (i.e. after the resource owner has been redirected back to the relying party)